### ISSUE TITLE
Add a page on why parametrized queries are better than escaping

### ISSUE THREAD

**Issue by petdance** [petdance/bobby-tables#46]

**Comment by MrStonedOne** [comment]
But, they aren't.

**Comment by andy-twosticks** [comment]
+1.  Comments like the one above demonstrate that this would be useful.

**Comment by MrStonedOne** [comment]
But, they aren't.

The argument is always that you can't forget to escape.

That's not even true, all it takes is one slip up and using `$name` rather than `:name` and you are vulnerable.

At least with escaping you can dynamically build the query text from conditions without having to either make massive repeating if else chains or maintaining what is basically a parallel array.

I know what I'm talking about when I saw that they are not objectively better: https://github.com/tgstation/tgstation13.org/blob/master/tgdb/irvpolltally.php#L60

**Comment by andy-twosticks** [comment]
I didn't say that they are perfect. They aren't.  But escaping is a highly flawed solutuion.

(re your example: no, it certainly won't save you if you try and parameterise a table name!  Bad idea.

Have a look through this slidedeck: http://www.slideshare.net/billkarwin/sql-injection-myths-and-fallacies

You might also read through the comments on this Bruce Schneier article, where a lot of very knowledable people break down the reasons why escaping is not the answer (at least, not on its own): https://www.schneier.com/blog/archives/2008/10/how_to_write_in.html

TL;DR: there is no magic bullet.  We are all vulnerable to a greater or lesser extent.

**Comment by petdance** [comment]
I should start a page that is just links to other articles and slide decks about SQL injection.

**Comment by MrStonedOne** [comment]
Linking me to some long ass article or some hard to quickly digest powerpoint is going to do nothing.

Explain what is actually wrong with escaping so I can smack your arguments down myself.

I've been programming in php for 16 years since i was 13, I've used both prepared queries and escaped queries. And right now I code for a open source website (and game server) where the primary user base is 4chan users who love sniffing out vulnerabilities and exploiting them for fun. And its escaped queries all the way down,

**Comment by MrStonedOne** [comment]
>(re your example: no, it certainly won't save you if you try and parameterise a table name! Bad idea.

My example is the chain of ifs that dynamically build the query text up, not the table name.

**Comment by petdance** [comment]
> Linking me to some long ass article or some hard to quickly digest powerpoint is going to do nothing.

No, not for you, for other users of the site.  Turns out I already started a ticket for this: #29

**Comment by MrStonedOne** [comment]
That was directed at andy.

**Comment by petdance** [comment]
> all it takes is one slip up and using $name rather than :name and you are vulnerable.

I hadn't thought of that.  That's a good argument for not using interpolation for your SQL strings, too.

**Comment by andy-twosticks** [comment]
@MrStonedOne 

> Explain what is actually wrong with escaping so I can smack your arguments down myself.

What a wonderful invitation, let me consider it .... nope.

Son, I'm not here to educate you and I don't enjoy arguing with people.  If you do, good luck with that.  Do your reading and learn something to your advantage.  Or don't.  All the same to me.

**Comment by MrStonedOne** [comment]
The point is you're wrong.

Neither is objectively better

**Comment by petdance** [comment]
> That was directed at andy.

I'm Andy, too.
